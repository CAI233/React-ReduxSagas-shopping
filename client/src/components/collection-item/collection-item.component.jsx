import React from 'react';

import CustomButtonExp from '../custom-button-exp/custom-button-exp.component';

import { connect } from 'react-redux';
import { addCartItem } from '../../redux/cart/cart.actions.js';

import { CollectionItemStyledContainer, Image,  CollectionFooter, Name, Price } from './collection-item.styles';

const CollectionItem = ({ item, addCartItem }) => {
    const {name, imageUrl, price } = item;
    return(
        <CollectionItemStyledContainer>
            <Image backgroundImg={imageUrl} >
                <CustomButtonExp 
                    className="collection-item-btn" 
                    isCollectionItemBtnStyles 
                    onClick={ ()=>addCartItem(item) } 
                >
                    加入购物车
                </CustomButtonExp>
            </Image>
            <CollectionFooter>
                <Name>{name}</Name>
                <Price>{price}￥</Price>
            </CollectionFooter>
        </CollectionItemStyledContainer>
    );
}

const mapDispatchToProps = dispatch => ({
    addCartItem: items => dispatch(addCartItem(items)),
});

export default connect(null,mapDispatchToProps)(CollectionItem);