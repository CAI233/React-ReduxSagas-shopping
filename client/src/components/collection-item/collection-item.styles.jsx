import styled,{css} from 'styled-components';
import { display_flex, xy_items, img_transition } from '../../assets/__OO7BTS.v1.0';


export const CollectionItemStyledContainer = styled.div`
    width: 20vw;
    display: flex;
    flex-direction: column;
    height: 525px;
    margin-bottom:0.4rem;
    @media ( max-width:1500px ) {
        width: 19vw;
        height:400px;
    }
    align-items: center;

    ${ img_transition(0.3) }    
    &:hover{
        opacity: 0.8;

        .collection-item-btn{
        display: block;
        }
    }  
`;
CollectionItemStyledContainer.displayName = "CollectionItemStyledContainer";

// 设定背景图片
const getBackgroundImg = props => {
    if( props.backgroundImg ){
        return css`
            background-image: url( ${props.backgroundImg});
        `;
    }
}
getBackgroundImg.displayName = "getBackgroundImg";

export const Image = styled.div`
    width: 100%;
    height: 95%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
    ${display_flex()}
    ${xy_items('center','flex-end')}
    ${getBackgroundImg}
`;
Image.displayName = "Image";

export const CollectionFooter = styled.div`
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-between;
    font-size: 18px;
`;

CollectionFooter.displayName = "CollectionFooter";

export const Name = styled.span`
    width: 90%;
    margin-bottom: 15px;
`;
Name.displayName = "Name";

export const Price = styled.span`
    width: 10%;
`;
Price.displayName = "Price";