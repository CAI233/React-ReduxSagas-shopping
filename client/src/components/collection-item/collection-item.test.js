import { shallow } from 'enzyme';
import React from 'react';
import { CollectionItem } from './collection-item.component';
import CustomButtonExp from '../custom-button-exp/custom-button-exp.component';

describe("测试单个商品显示",()=>{
    let wrapper,mockProps;
    const mockItem = {
        imageUrl: '1.jpg',
        name: 'test',
        price: 10,
        quantity: 1,
    };
    beforeEach(()=>{
        mockProps ={
            item: mockItem,
            addCartItem: jest.fn(),
        }
        wrapper = shallow(<CollectionItem {...mockProps} />);
    });
   
    it(" CollectionItem Snapshot ",()=>{
        expect(wrapper).toMatchSnapshot();
    });

    it(" 测试加入购物车功能 ",()=>{
        wrapper.find(CustomButtonExp).simulate('click');
        expect( mockProps.addCartItem ).toHaveBeenCalled();
    });

    it(" 测试图片链接是否到位 ",()=>{
        expect( wrapper.find("Image").prop('backgroundImg') ).toEqual( mockItem.imageUrl ); // prop(): 获取组件传入的值( 等待笔记 )
    });

});