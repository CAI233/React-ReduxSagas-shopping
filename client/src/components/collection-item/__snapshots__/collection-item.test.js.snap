// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试单个商品显示  CollectionItem Snapshot  1`] = `ShallowWrapper {}`;
